### DISCUSSION TITLE
rAudio-1 running on BeagleBone Black

### DISCUSSION THREAD

**Discussion by fillods** [rern/rAudio#299] [Tips]
First of all, rAudio-1 works on BeagleBone Black \o/

Here is how I proceeded:
1. Install latest ArchLinux ARM on a µSD card as guided by 
  https://archlinuxarm.org/platforms/armv7/ti/beaglebone-black#installation
  Rem: I did not use the [rOS create.sh script](https://github.com/rern/rOS/raw/main/create.sh) which is too much RPi centric.
  However reading it and the start page of https://github.com/rern/rOS gave me good insight.
  Please note  BBB's ArchLinux has/needs no BOOT partition.
2. Perform manually some of the commands inspired by the [rOS create-alarm.sh script](https://github.com/rern/rOS/raw/main/create-alarm.sh).
  Out of my head:
  - `echo noarp >> $ROOT/etc/dhcpcd.conf`
  - `echo DNSSEC=no >> $ROOT/etc/systemd/resolved.conf`
  - `rm -r $ROOT/etc/systemd/system/network-online.target.wants`
  - disable root passwd (really needed? will be reactivated later), enable SSH root login
  - `systemctl start systemd-random-seed`
  - `echo bluez bluez-alsa bluez-utils python-dbus python-gobject python-requests kid3-common samba spotifyd upmpdcli > $ROOT/boot/features`
3. forge a `$ROOT/boot/versions` file:
```
1
20210402
main
139
0
```
4. create a dummy file `$ROOT/boot/config.txt`, many rAudio scripts will grep it
5. UTF-8 support on the target:
  - enable `en_GB.UTF-8` in `/etc/locale.gen`
  - run `locale-gen`
  - `sudo localectl set-locale en_GB.UTF-8`
6. on the target download and run `/root/create-ros.sh` as root
7. patch rAudio to be less Raspberry-Pi centric, see this pull request https://github.com/rern/rAudio-1/pull/288
8. Check how it's starting up: `journalctl -e --no-hostname`, fix issues as they come..

Configuration has been done through the web interface. Right now the DAC is an USB audio device, to be improved later with an I²S audio module. Beside others, I use gmpc to control rAudio.
_I love the FIP (webradio) add-on !_

Next step is making IR receiver/LIRC to work with my remote. It's kind of troublesome on BBB since there's [no more bone_capemgr in recent kernel](https://archlinuxarm.org/forum/viewtopic.php?f=23&t=12635) to declare the GPIO. It has to be done by U-Boot. More to come on that later.

**Comment by fillods** [comment]
The afore mentioned pull request #288 has been integrated in [release 20210414](https://github.com/rern/rAudio-1/releases/tag/20210414). Yay !

![BBB system 20210414](https://user-images.githubusercontent.com/809755/114844059-ba5cb000-9dda-11eb-9887-bcf6cd4794bd.png)

The loadavg is quite high, though. It looks like something (a script? status.sh ?) is called on a loop. To be investigated.

**Reply by CastleCaptain** [reply]
Not a coder but these are the things I had to do to get it working on a BBB.
1. ArchLinuxArm mirror error when running create-ros.sh 
	fi
	file=/etc/pacman.d/mirrorlist
	prevmirror=$( grep ^Server $file \
					| head -1 \
					| sed 's|\.*mirror.*||; s|.*//||' )
	if [[ $mirror != $prevmirror ]]; then
		if [[ $mirror == 0 ]]; then
			mirror=
			rm $dirsystem/mirror
		else
			echo $mirror > $dirsystem/mirror
			mirror+=.
		fi
		sed -i "0,/^Server/ s|//.*mirror|//${mirror}mirror|" $file
	fi
Workaround was to manually specify mirror in /etc/pacman.d/mirrorlist

2.  Main and $release directories were not downloading resulting in a bsdtar errors when running create-rosh.sh
curl -skLO https://github.com/rern/rOS/archive/main.tar.gz
curl -skLO https://github.com/rern/rAudio-$version/archive/$release.tar.gz
mkdir -p /tmp/config
bsdtar --strip 1 -C /tmp/config -xvf main.tar.gz
bsdtar --strip 1 -C /tmp/config -xvf $release.tar.gz
Workaround was to change $version to 1 and $release to 202203271 prior to running create-ros.sh 

3. Was getting a taskset error when starting mpd.service. Solution was to change the taskset from 3 to 0 in /etc/systemd/system/mpd.service.d/override.conf
